### PR TITLE
Sensor: mpr: Add Honeywell MPR driver and example

### PR DESCRIPTION
**Pull request by sven-hm**
The Honeywell MPR sensor is a piezoresistive silicon pressure sensor,
which can be accessed via i2c.

https://sensing.honeywell.com/micropressure-mpr-series

Signed-off-by: Sven Herrmann <sven.herrmann@posteo.de>

### CODE REVIEW

File: drivers/sensor/mpr/mpr.c [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	return 0;
+}
+
+static const struct sensor_driver_api mpr_api_funcs = {
+	.sample_fetch = mpr_sample_fetch,
+	.channel_get = mpr_channel_get,
+};
+
+static struct mpr_data mpr_data;
+static const struct mpr_config mpr_cfg = {
```
Comment (galak) [inline drivers/sensor/mpr/mpr.c:128]: Please use the new DT_INST macro's in include/devicetree.h (see example in drivers/sensor/fxos8700)

File: dts/bindings/sensor/honeywell,mpr.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+
+description: Honeywell Digital Pressure Sensor
+
```
Comment (galak) [inline dts/bindings/sensor/honeywell,mpr.yaml:4]: Is there a datasheet URL or something you can possible reference here?

**Review by galak — COMMENTED** [review]
You'll need to add a dts node in tests/drivers/build_all/i2c.dtsi

File: dts/bindings/sensor/honeywell,mpr.yaml (hunk 2 above)
Comment (sven-hm) [reply dts/bindings/sensor/honeywell,mpr.yaml:6]: Ok, added.

File: drivers/sensor/mpr/mpr.c (hunk 1 above)
Comment (sven-hm) [reply drivers/sensor/mpr/mpr.c:128]: Ok, done.

File: drivers/sensor/mpr/Kconfig [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+config MPR_OUTPUT_TYPE_3
+	bool "i2c, address 0x38"
+config MPR_OUTPUT_TYPE_4
+	bool "i2c, address 0x48"
+config MPR_OUTPUT_TYPE_5
+	bool "i2c, address 0x58"
+config MPR_OUTPUT_TYPE_6
+	bool "i2c, address 0x68"
+config MPR_OUTPUT_TYPE_7
+	bool "i2c, address 0x78"
```
Comment (MaureenHelm) [inline drivers/sensor/mpr/Kconfig]: this should come from device tree, not kconfig

File: drivers/sensor/mpr/mpr.c [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+#include <drivers/i2c.h>
+#include <sys/byteorder.h>
+#include <init.h>
+#include <logging/log.h>
+#include "mpr.h"
+#include "mpr_configuration.h"
+
+LOG_MODULE_REGISTER(MPR, CONFIG_SENSOR_LOG_LEVEL);
+
+int mpr_init(struct device *dev)
```
Comment (MaureenHelm) [inline drivers/sensor/mpr/mpr.c:24]: make this `static`

File: drivers/sensor/mpr/mpr.c [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+{
+	struct mpr_data *data = dev->driver_data;
+	const struct mpr_config *cfg = dev->config->config_info;
+
+	data->i2c_master = device_get_binding(cfg->i2c_bus);
+	if (!data->i2c_master) {
+		LOG_DBG("mpr: i2c master not found: %s", cfg->i2c_bus);
```
Comment (MaureenHelm) [inline drivers/sensor/mpr/mpr.c:31]: replace `LOG_DBG` with `LOG_ERR`

File: samples/sensor/mpr/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
```
Comment (MaureenHelm) [inline samples/sensor/mpr/CMakeLists.txt:3]: please move the sample to a separate commit. you can keep it in the same PR though

File: samples/sensor/mpr/sample.yaml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+sample:
+  name: MPR Pressure Sensor
+tests:
+  sample.sensor.mpr:
+    harness: sensor
+    tags: sensors
+    depends_on: i2c
+    filter: dt_compat_enabled("honeywell,mpr")
```
Comment (MaureenHelm) [inline samples/sensor/mpr/sample.yaml:8]: we don't have any boards with this compatible, so this sample isn't going to be built in CI. i usually don't like whitelists, but in this case i think you should add arduino_due as a platform_whitelist instead of the filter.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
thanks for the contribution!

please squash the fixup commit into the previous commit, and rebase to the latest master to fix the merge conflict.

File: drivers/sensor/mpr/Kconfig (hunk 3 above)
Comment (sven-hm) [reply drivers/sensor/mpr/Kconfig]: Ok, removed.

File: drivers/sensor/mpr/mpr.c (hunk 4 above)
Comment (sven-hm) [reply drivers/sensor/mpr/mpr.c]: Ok, done.

File: drivers/sensor/mpr/mpr.c (hunk 5 above)
Comment (sven-hm) [reply drivers/sensor/mpr/mpr.c]: Ok, done.

File: samples/sensor/mpr/CMakeLists.txt (hunk 6 above)
Comment (sven-hm) [reply samples/sensor/mpr/CMakeLists.txt:3]: Ok, done.

File: samples/sensor/mpr/sample.yaml (hunk 7 above)
Comment (sven-hm) [reply samples/sensor/mpr/sample.yaml]: Ok, done.

**Comment by sven-hm** [comment]
The PR is now split into two commits, one for the driver the other for the sample. I cleaned up the header and Kconfig files and added a node in tests/drivers/build_all/i2c.dtsi.

**Review by MaureenHelm — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
